paths_cli/compiling/engines.py

```diff
@@ -0,0 +1,54 @@
+from .topology import build_topology
+from .core import Builder
+from paths_cli.compiling.core import Parameter
+from .tools import custom_eval_int
+from paths_cli.compiling.plugins import EngineCompilerPlugin, CategoryPlugin
+
+try:
+    from simtk import openmm as mm
+except ImportError:
+    HAS_OPENMM = False
+else:
+    HAS_OPENMM = True
+
+def load_openmm_xml(filename):
+    if not HAS_OPENMM:  # -no-cov-
+        raise RuntimeError("OpenMM does not seem to be installed")
+
+    with open(filename, mode='r') as f:
+        obj = mm.XmlSerializer.deserialize(f.read())
+
+    return obj
+
+def openmm_options(dct):
+    n_steps_per_frame = dct.pop('n_steps_per_frame')
+    n_frames_max = dct.pop('n_frames_max')
+    options = {'n_steps_per_frame': n_steps_per_frame,
+               'n_frames_max': n_frames_max}
+    dct['options'] = options
+    return dct
```

Comment:
This is used as a `remapper` in the OpenMM engine `Builder`. The `Builder` calls `remapper` on a copy of the relevant `dict`, so the question of running twice isn't an issue. I also made remapper functions private. They really only exist so they can be composed as part of the `Builder`, and I don't think there are any that should be re-used outside of that context.

It might be reasonable to provide a better error message than the `KeyError` that would occur here, but I'm not doing that now. It feels to me like that's a more generic problem that should be caught higher up in the stack.